src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -442,4 +565,125 @@ private Model GetVocabularyDictionary()
             }
         }
     }
+
+    public sealed class WordEmbeddingsEstimator : TrivialEstimator<WordEmbeddingsTransform>
+    {
+        public WordEmbeddingsEstimator(IHostEnvironment env, string inputColumn, string outputColumn,
+           WordEmbeddingsTransform.PretrainedModelKind modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe)
+            : this(env, new WordEmbeddingsTransform(env, inputColumn, outputColumn, modelKind))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env, string inputColumn, string outputColumn, string customLookupTable)
+            : this(env, new WordEmbeddingsTransform(env, inputColumn, outputColumn, customLookupTable))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env,
+            WordEmbeddingsTransform.PretrainedModelKind modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe, params WordEmbeddingsTransform.ColumnInfo[] columns)
+            : this(env, new WordEmbeddingsTransform(env, modelKind, columns))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env, string customLookupTable, params WordEmbeddingsTransform.ColumnInfo[] columns)
+            : this(env, new WordEmbeddingsTransform(env, customLookupTable, columns))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env, WordEmbeddingsTransform transformer)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(WordEmbeddingsEstimator)), transformer)
+        {
+        }
+
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            foreach (var colInfo in Transformer.Columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.input, out var col))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.input);
+                if (!(col.ItemType is TextType) || (col.Kind != SchemaShape.Column.VectorKind.VariableVector && col.Kind != SchemaShape.Column.VectorKind.Vector))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.input, new VectorType(TextType.Instance).ToString(), col.GetTypeString());
+
+                result[colInfo.output] = new SchemaShape.Column(colInfo.output, SchemaShape.Column.VectorKind.Vector, NumberType.R4, false);
+            }
+
+            return new SchemaShape(result.Values);
+        }
+    }
+
+    public static class WordEmbeddingsStaticExtensions
+    {
+        /// <include file='doc.xml' path='doc/members/member[@name="WordEmbeddings"]/*' />
```

Comment:
><include file='doc.xml' path='doc/members/member[@name="WordEmbeddings"]/*' /> [](start = 12, length = 78)

While linking this is probably nice, we should probably also have parameter descriptors, should we not?

Reply:
I thought about it too but worried about if its rendered properly. Let me talk to @Senja.

---
In reply to: [218234747](https://github.com/dotnet/machinelearning/pull/928#discussion_r218234747) [](ancestors = 218234747)

Reply:
Talking to our local expert @sfilipi is never wasted, but we could also try it out... maybe from the test assembly, hit the dots, see what shows up in intellisense. Based on my (almost certainly imperfect) understanding, I *think* any defined elements are just smooshed together in the output, so even while we can get the import to get the summary and remarks, any params would be rendered without complaint.

I do wonder however what stylecop will make of all that though...

---
In reply to: [218236334](https://github.com/dotnet/machinelearning/pull/928#discussion_r218236334) [](ancestors = 218236334,218234747)